src/Tools/a2md/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace a2md
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            Loader loader = new Loader();
+            DescriptorEqualityComparer comparer = new DescriptorEqualityComparer();
+
+            var outputMarkdown = args
+                .Select(arg => new AnalyzerFileReference(arg, loader))
+                .SelectMany(analyzerReference => analyzerReference.GetAnalyzersForAllLanguages())
+                .SelectMany(analyzer => analyzer.SupportedDiagnostics)
+                .Distinct(comparer)
```

Comment:
It's not a big concern. In general, re-using IDs across `DiagnosticDescriptor`s with different titles/descriptions/messages is a bad idea, but for UX rather than functional reasons. For example, you can't control the severity of those diagnostics individually, since that's a per-ID setting. Also, Solution Explorer only ever shows one descriptor per ID; we decided that showing multiple would be too confusing. This tools follows the same logic and just picks one, though I could go either way on that.

FxCop actually did this with some of its rules. For example, I think there are three rules around the use of `FlagsAttribute` on enum types that all have the same ID but fire under slightly different circumstances. When we reimplemented those rules as Roslyn analyzers we retained the duplication.